core/mvcc/database/tests.rs

```diff
@@ -524,20 +510,18 @@ fn test_lost_update() {
     let conn2 = db.db.connect().unwrap();
     let tx2 = db.mvcc_store.begin_tx(conn2.pager.borrow().clone());
     let tx2_row = generate_simple_string_row(1, 1, "World");
-    assert!(db
-        .mvcc_store
-        .update(tx2, tx2_row.clone(), conn2.pager.borrow().clone())
-        .unwrap());
+    assert!(db.mvcc_store.update(tx2, tx2_row.clone()).unwrap());
 
     // T3 also attempts to update row ID 1 within an active transaction.
     let conn3 = db.db.connect().unwrap();
     let tx3 = db.mvcc_store.begin_tx(conn3.pager.borrow().clone());
     let tx3_row = generate_simple_string_row(1, 1, "Hello, world!");
     assert!(matches!(
-        db.mvcc_store
-            .update(tx3, tx3_row, conn3.pager.borrow().clone(),),
+        db.mvcc_store.update(tx3, tx3_row),
         Err(LimboError::WriteWriteConflict)
     ));
+    // hack: in the actual tursodb database we rollback the mvcc tx ourselves, so manually roll it back here
```

Comment:
if we want to avoid this, then we need to add some `mv_tx_was_already_rolled_back_internally_lmao` flag to `mv_store.rollback_tx()` function so that we don't double-rollback and crash